internal/users/manager.go

```diff
@@ -33,15 +37,55 @@ var DefaultConfig = Config{
 	GIDMax: 1999999999,
 }
 
+type temporaryUser struct {
+	name    string
+	preAuth bool
+}
+
 // Manager is the manager for any user related operation.
 type Manager struct {
```

Comment:
The change to the Manager seems to be big enough that it would make sense to move the new code to a new struct that the manager includes to me.

So that we can have all the related code in a separated file